web/pages/Chub/ChubNavigation.tsx

```diff
@@ -0,0 +1,84 @@
+import { Component, Show, createEffect, createSignal } from 'solid-js'
+import { chubStore } from '../../store/chub'
+import { chubOptions } from './Chub'
+import TextInput from '../../shared/TextInput'
+import Button from '../../shared/Button'
+import { ArrowLeft, ArrowRight } from 'lucide-solid'
+import { toastStore } from '../../store'
+
+export const [chubPage, setChubPage] = createSignal<number>(1)
+
+const ChubNavigation: Component<{ buttons: boolean }> = (props) => {
+  createEffect(() => {
+    chubStore.getChubChars()
+  })
+
+  return (
+    <>
+      <div class="mt-2 flex justify-between">
+        <div class="flex gap-2">
+          <TextInput
+            fieldName="search"
+            placeholder="Search by name..."
+            value={chubOptions.search}
+            onKeyUp={(ev) => {
+              chubOptions.search = ev.currentTarget.value
```

Comment:
We definitely want to avoid these styles of mutations. We should stick with idiomatic state management (chubStore or signals, whichever suits best).